examples/imm_example2.py

```diff
@@ -0,0 +1,287 @@
+    Returns
+    -------
+        A matplotlib ellipse artist
+    """
+
+    def eigsorted(cov):
+        vals, vecs = np.linalg.eigh(cov)
+        order = vals.argsort()[::-1]
+        return vals[order], vecs[:, order]
+
+    if ax is None:
+        ax = plt.gca()
+
+    vals, vecs = eigsorted(cov)
+    theta = np.degrees(np.arctan2(*vecs[:, 0][::-1]))
+
+    # Width and height are "full" widths, not radius
+    width, height = 2 * nstd * np.sqrt(vals)
+    ellip = Ellipse(xy=pos, width=width, height=height, angle=theta,
+                    **kwargs)
+
+    ax.add_artist(ellip)
+    return ellip
+
+##############################################################################
+# MODELS                                                                     #
+##############################################################################
+transition_model_1 = CombinedLinearGaussianTransitionModel(
+                        (OrnsteinUhlenbeck(0.1 ** 2, 2e-2),
+                         OrnsteinUhlenbeck(0.1 ** 2, 2e-2)))
+transition_model_2 = CombinedLinearGaussianTransitionModel(
+                        (RandomWalk(0.0000000000000001**2),
+                         RandomWalk(np.finfo(float).eps),
+                         RandomWalk(0.0000000000000001**2),
+                         RandomWalk(np.finfo(float).eps)))
+trans_1 = CombinedLinearGaussianTransitionModel((ConstantVelocity(0.1),
+                                                ConstantVelocity(0.1)))
+trans_2 = CombinedLinearGaussianTransitionModel((ConstantAcceleration(0.1),
+                                                ConstantAcceleration(0.1)))
+sim_trans_model = trans_1
+
+measurement_model = LinearGaussian(ndim_state=4, mapping=[0, 2],
+                                   noise_covar=np.diag([1 ** 2,
+                                                        1 ** 2]))
+
+measurement_model2 = LinearGaussian(ndim_state=6, mapping=[0, 3],
+                                    noise_covar=np.diag([1 ** 2,
+                                                         1 ** 2]))
+##############################################################################
+# PREDICTOR/UPDATER                                                          #
+##############################################################################
+model_transition_matrix = np.array([[0.95, 0.05],
+                                    [0.3, 0.7]])
+
+KalmanPredictorMM = add_mixture_capability(KalmanPredictor)
+predictor_1 = KalmanPredictor(transition_model_1)
+predictor_2 = KalmanPredictor(transition_model_2)
+predictor_cv = KalmanPredictorMM(trans_1, convert2common_state=cv2ca,
+                                 convert2local_state=ca2cv)
+predictor_ca = KalmanPredictorMM(trans_2, convert2common_state=null_convert,
+                                 convert2local_state=null_convert)
+imm_predictor = IMMPredictor([predictor_cv, predictor_ca],
+                             model_transition_matrix)
```

Comment:
Would it be possible to convert the IMMPredictor to take a filter class, motion models and transition matrix as inputs (with optional input for convert2common_state and convert2local_state). Internally the IMMPredictor could then construct the set of filters from its inputs. 

At the moment their is a significant burden on the user to build up these banks of filters and it requires significant knowledge of the code base. Something like the below would be easier for user pickup

``imm_predictor = IMMPredictor(KalmanPredictor, [transition_model_1, transition_model_2], model_transition_matrix)

internally the IMMPredictor could then check the transition models to build the appropriate filter class MM extensions